### PR TITLE
[Merged by Bors] - Logging

### PR DESCRIPTION
**Pull request by siegfriedweber**
## Description

- Show opensearch-dashboards service with `stackablectl services list`.
- Add logging demo.

TODOs before merging:
- [x] `demos/demos-v1.yaml`: Replace `https://raw.githubusercontent.com/stackabletech/stackablectl/logging/demos/logging/zookeeper.yaml` with `https://raw.githubusercontent.com/stackabletech/stackablectl/main/demos/logging/zookeeper.yaml`.
- [x] `stacks/stacks-v1.yaml`: Replace `https://raw.githubusercontent.com/stackabletech/stackablectl/logging/stacks/logging/vector-aggregator-discovery.yaml` with `
https://raw.githubusercontent.com/stackabletech/stackablectl/main/stacks/logging/vector-aggregator-discovery.yaml`.
- [x] `demos/logging/zookeeper.yaml`: Remove `repo: docker.stackable.tech/sandbox/logging` when stackabletech/docker-images#268 is merged and the images are built and deployed.
- [x] `releases/releases.yaml`: Remove the file when stackabletech/zookeeper-operator#588 is merged.
- [x] `stacks/stacks-v1.yaml`: Replace `stackableRelease: 22.11-logging` with `stackableRelease: 22.11` when stackabletech/zookeeper-operator#588 is merged.

## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: docs/modules/ROOT/nav.adoc [hunk 1]
```diff
@@ -10,6 +10,7 @@
 ** xref:demos/airflow-scheduled-job.adoc[]
 ** xref:demos/data-lakehouse-iceberg-trino-spark.adoc[]
 ** xref:demos/hbase-hdfs-load-cycling-data.adoc[]
+** xref:demos/logging.adoc[]
```
Comment (sbernauer) [inline docs/modules/ROOT/nav.adoc:13]: Not sure where to put this docs. Fine for me for now, as long as we have not decided how to ship the whole logging stack.

**Review by fhennig — CHANGES_REQUESTED** [review]
I've only read this docs page for now, and I have this one comment.

i've ran the demo on my machine and it worked!

File: src/services.rs [hunk 2]
```diff
@@ -89,6 +89,14 @@ lazy_static! {
                     kind: "OpaCluster".to_string(),
                 }
             ),
+            (
```
Comment (sbernauer) [inline src/services.rs:92]: We can get rid of this as we just list services, can't we?

File: stacks/stacks-v1.yaml [hunk 3]
```diff
@@ -164,7 +164,226 @@ _templates:
...
+                    secrets.stackable.tech/scope: |-
+                      service=opensearch-cluster-master
+                spec:
+                  storageClassName: secrets.stackable.tech
+                  accessModes:
+                    - ReadWriteOnce
+                  resources:
+                    requests:
+                      storage: 1
+        extraObjects:
```
Comment (sbernauer) [inline stacks/stacks-v1.yaml:232]: We can also throw in the CM `vector-aggregator-discovery` here. Would make this thing more self-contained

**Comment by sbernauer** [comment]
Demo worked for me as well

**Comment by sbernauer** [comment]
image
my usual problems with elastic/Opensource (ordering of lines) :upside_down_face: But nothing we can do here afaik

File: docs/modules/ROOT/nav.adoc (hunk 1 above)
Comment (siegfriedweber) [reply docs/modules/ROOT/nav.adoc:13]: This is not the documentation for the stack. It is the documentation for the demo, so it should be listed together with the other demo documentations, shouldn't it?

File: src/services.rs (hunk 2 above)
Comment (siegfriedweber) [reply src/services.rs]: Yes, we can. This was a leftover of a prior implementation.

File: stacks/stacks-v1.yaml (hunk 3 above)
Comment (sbernauer) [reply stacks/stacks-v1.yaml:232]: Agreed, missed that where are at the opensearch chart here

File: docs/modules/ROOT/nav.adoc (hunk 1 above)
Comment (sbernauer) [reply docs/modules/ROOT/nav.adoc:13]: Yes :)

**Review by sbernauer — APPROVED** [review]
Please wait for Felix

File: docs/modules/ROOT/pages/demos/logging.adoc [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+least 262144.
+
+[NOTE]
+====
+If you are on a Mac, the commands in this section won’t work for you. You can
+skip it, and continue with the rest of the tutorial. They are not strictly
+necessary to get a working prototype going.
+====
+
+To check the current value, run this command:
```
Comment (fhennig) [inline docs/modules/ROOT/pages/demos/logging.adoc:39]:
```suggestion
== OpenSearch prerequisites

=== Mac and Windows
 If you're running on Mac or Windows and use Docker to run
Kubernetes, in menu:Preferences[Resources] set the RAM to at least 4GB.

=== Linux

OpenSearch uses a mmapfs directory by default to store its indices. The default
operating system limits on mmap counts is likely to be too low – usually 65530,
which may result in out of memory exceptions. So the Linux setting
`vm.max_map_count` on the host machine where kind is running, must be set to at
least 262144.

To check the current value, run this command:
```

Comment (fhennig) [reply docs/modules/ROOT/pages/demos/logging.adoc:39]: I'd do it like this

Comment (siegfriedweber) [reply docs/modules/ROOT/pages/demos/logging.adoc]: adopted

**Review by fhennig — APPROVED** [review]
I'm happy now :smiley: LGTM!

**Review by sbernauer — APPROVED** [review]
Happy as well

**Comment by siegfriedweber** [comment]
bors merge

**Pull request closed without merging**